clidocstool.go

```diff
@@ -77,9 +87,69 @@ func (c *Client) GenAllTree() error {
 	if err = c.GenYamlTree(c.root); err != nil {
 		return err
 	}
+	if err = c.GenManTree(c.root); err != nil {
+		return err
+	}
 	return nil
 }
 
+// loadLongDescription gets long descriptions and examples from markdown.
+func (c *Client) loadLongDescription(parentCmd *cobra.Command, generator string) error {
+	for _, cmd := range parentCmd.Commands() {
+		if cmd.HasSubCommands() {
+			if err := c.loadLongDescription(cmd, generator); err != nil {
```

Comment:
Is the intent to use the same markdown files for man pages and online docs?

I know we had some discussions about that in the past, and while this would be "great!", it can be somewhat difficult (and I wasn't able to come up with a one-size-fits-all solution). Conventions in man pages can be quite different from conventions in online docs. For example, the online docs would refer to another command using something like "for details on `other command`, refer to the `<link>other command reference documentation</link>`", whereas in a man page it would be "see `other-command(1)`".


For the cli, that lead us to maintain two separate sets of markdown files (https://github.com/docker/cli/tree/3fb4fb83dfb5db0c0753a8316f21aea54dab32c5/man), but that situation is also kinda horrible, as they're not as well maintained, and often not as up-to-date as the equivalent "online docs" man pages.

Reply:
Hum yeah it will be hard to be generic with the library. I will think about it and keep you in touch.

Reply:
I wonder if https://github.com/muesli/mango could solve this issue. I will take a look.